src/main/java/gov/nasa/cumulus/metadata/aggregator/UMMGranuleFile.java

```diff
@@ -450,10 +450,13 @@ private JSONObject exportSpatial() throws ParseException{
             if(this.isoXMLSpatialTypeEnumHashSet.contains(MENDsIsoXMLSpatialTypeEnum.FOOTPRINT)) {
                 AdapterLogger.LogDebug(this.className + "UMMGranuleFile.exportSpatial FOOTPRINT Processing");
                 String polygon = ((IsoGranule) granule).getPolygon();
-                if (polygon != "" && polygon != null) {
-                    // Export Polygon
-                    addPolygon(geometry, polygon);
-                }
+                AdapterLogger.LogInfo(this.className + " nc.iso.xml footprint processing ... ");
+                this.isLineFormattedPolygon = true;
+                geometry = line2Polygons(geometry,polygon);
```

Comment:
Does `line2Polygons` divide over IDL? What happens if the geometry isn't a line but is a polygon?

Reply:
Yes, line2Polygons divide over IDL. 
I guess the name of the function is not clear.  It is expecting a either a line or polygon, i.e. both will work.

Some of the s6 posList coordinates are line, the code will add lon, lat in the end to make it a "close loop of polygon".  If the input coordinate list is already a polygon, it will try to divide over IDL (if over IDL)